conans/model/build_info.py

```diff
@@ -209,15 +210,49 @@ def name(self):
     def name(self, value):
         self._name = value
 
+    # TODO: Deprecate for 2.0. Only cmake and pkg_config generators should access this.
+    #  Use get_property for 2.0
     def get_name(self, generator):
-        return self.names.get(generator, self._name)
+        property_name = "cmake_target_name" if "cmake" in generator else "pkg_config_name"
+        return self.get_property(property_name, generator) or self.names.get(generator, self._name)
 
+    # TODO: Deprecate for 2.0. Only cmake generators should access this. Use get_property for 2.0
     def get_filename(self, generator):
-        result = self.filenames.get(generator)
+        result = self.get_property("cmake_file_name", generator) or self.filenames.get(generator)
         if result:
             return result
         return self.get_name(generator)
 
+    # TODO: Deprecate for 2.0. Use get_property for 2.0
+    def get_build_modules(self):
+        default_values_dict = self._generator_properties.get("conan_default_generators_value")
+        default_build_modules_value = default_values_dict.get("cmake_build_modules") if default_values_dict else None
+        ret_dict = {"cmake_find_package": default_build_modules_value,
+                    "cmake_find_package_multi": default_build_modules_value,
+                    "cmake": default_build_modules_value,
+                    "cmake_multi": default_build_modules_value} if default_build_modules_value else {}
+        for generator, values in self._generator_properties.items():
+            if values.get("cmake_build_modules") and generator != "conan_default_generators_value":
+                ret_dict[generator] = values.get("cmake_build_modules")
+        return ret_dict if ret_dict else self.build_modules
+
+    def set_property(self, property_name, value, generator=None):
+        generator = generator or "conan_default_generators_value"
+        gen_dict = self._generator_properties.get(generator)
+        if gen_dict:
+            gen_dict.update({property_name: value})
+        else:
+            self._generator_properties.update({generator: {property_name: value}})
+
+    def get_property(self, property_name, generator=None):
+        generator = generator or "conan_default_generators_value"
+        gen_dict = self._generator_properties.get(generator)
+        if gen_dict and gen_dict.get(property_name):
+            return gen_dict.get(property_name)
+        else:
+            gen_dict = self._generator_properties.get("conan_default_generators_value")
+            return gen_dict.get(property_name) if gen_dict else None
+
```

Comment:
```suggestion
        if generator:
            try:
                return self._generator_properties[generator][property_name]
            except KeyError:
                pass
        try:
            return self._generator_properties[None][property_name]
        except KeyError:
            pass
```

- avoid getting twice ``gen_dict.get(property_name)``
- There is a redundant double processing when ``generator=None``, same thing is being evaluated twice.